src/init.ts

```diff
@@ -0,0 +1,85 @@
+import { JaaS_DOMAIN } from "./constants";
+
+type ExternalApiMap = {
+  [key: string]: {
+    fn?: any;
+    isLoaded: boolean;
+    callbacks: ((err: Error | null, jitsiClass?: any) => void)[];
+    err: Error | null;
+  };
+};
+
+const externalApiMap: ExternalApiMap = {};
+
+/**
+ * Injects the external_api.js script for the corresponding domain in DOM
+ * and resolves with an object containing either the `JitsiMeetExternalAPI`
+ * class definition or an error
+ *
+ * @param {string} domain - The domain of the external API
+ * @returns {Promise<object>} - Object containing the JitsiMeetExternalAPI object
+ * or an error
+ */
+export const initExternal = (domain: string = JaaS_DOMAIN): Promise<void> =>
+  new Promise((resolve, reject) => {
+    init(domain, (error: Error | null, jitsiClass?: any): void => {
+      if (error) {
+        reject(error);
+      }
+      resolve(jitsiClass);
+    });
+  });
+
+export const init = (
+  domain: string,
+  callback: (err: Error | null, jitsiClass?: any) => void
+): void => {
+  if (!externalApiMap[domain]) {
+    externalApiMap[domain] = {
+      isLoaded: false,
+      callbacks: [callback],
+      err: null,
+    };
+    const script: HTMLScriptElement = document.createElement("script");
+    script.async = true;
+    script.src = `https://${domain}/external_api.js`;
+    script.onload = () => {
+      externalApiMap[domain].isLoaded = true;
+      externalApiMap[domain].fn = window.JitsiMeetExternalAPI.bind({});
```

Comment:
Why are we wrapping it like this, instead of returning `window.JitsiMeetExternalAPI` directly?

Reply:
Me and @mihhu  brainstormed this part, we did all this mapping and cloning of the JitsiMeetExternalAPI class since components can be on different domains. While generally the external_api.js can be used inter-domains, there are cases in which on one domain we have one or more commands/functions/events which are not found on external_api from other domain.

Reply:
This has never happened IRL, so let's not over-complicate things unless we absolutely need to, which is not the case here :-)